libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -14,35 +14,52 @@ import {
   BucketPutOptions,
   BucketDeleteOptions,
 } from "../cloud";
+import { deserialize, serialize } from "../simulator/serialization";
 import {
   ISimulatorContext,
   ISimulatorResourceInstance,
 } from "../simulator/simulator";
 import { Datetime, Json } from "../std";
 
+const METADATA_FILENAME = "metadata.json";
+
 export class Bucket implements IBucketClient, ISimulatorResourceInstance {
-  private readonly objectKeys: Set<string>;
   private readonly _fileDir: string;
   private readonly context: ISimulatorContext;
   private readonly initialObjects: Record<string, string>;
   private readonly _public: boolean;
   private readonly topicHandlers: Partial<Record<BucketEventType, string>>;
-  private readonly _metadata: Record<string, ObjectMetadata> = {};
+  private _metadata: Map<string, ObjectMetadata>;
 
   public constructor(props: BucketSchema["props"], context: ISimulatorContext) {
-    this.objectKeys = new Set();
-    this._fileDir = fs.mkdtempSync(join(os.tmpdir(), "wing-sim-"));
+    this._fileDir = join(context.statedir, "files");
     this.context = context;
     this.initialObjects = props.initialObjects ?? {};
     this._public = props.public ?? false;
     this.topicHandlers = props.topics;
-  }
-
-  public get fileDir(): string {
-    return this._fileDir;
+    this._metadata = new Map();
   }
 
   public async init(): Promise<BucketAttributes> {
+    const fileDirExists = await exists(this._fileDir);
+    if (!fileDirExists) {
+      await fs.promises.mkdir(this._fileDir, { recursive: true });
+    }
+
+    if (fs.existsSync(join(this.context.statedir, METADATA_FILENAME))) {
```

Comment:
Mixing sync and async fs calls on purpose?

Reply:
ah no, just result of some weird debugging rabbit holes - will fix :)